smurfs/src/components/Forms.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux';
+import { addSmurf } from '../actions';
+
+class Forms extends Component {
+
+state={
+  smurf:{
+    name:'',
+    age:'',
+    height:''
+  }
+}
+
+componentDidMount(){
+  this.props.addSmurf()
+}
+
+
+
+
+handleChange = e => {
+  console.log(e.target.name, e.target.value, this.state.smurf.name)
+  this.setState({
+    [e.target.name]:e.target.value
```

Comment:
Watch what happens in your state when you type into the input fields. Check the react dev tools and try typing into the fields. Note what happens and see if you can fix it! You can go back to the codesandbox from yesterday, it may hold some hints